src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -12,7 +12,7 @@ import {useFeatures} from '../../utilities/features';
 
 type OriginalColorScheme = Required<ThemeConfig['colorScheme']>;
 type Inverse = 'inverse';
-type InversableColorScheme = OriginalColorScheme | Inverse;
+export type InversableColorScheme = OriginalColorScheme | Inverse;
```

Comment:
Is it OK to put the export here or should it live somewhere else?

Reply:
I think if you put it here it's public. Can you import it without that?

Reply:
I need to import it in the other components that accept a `colorScheme` with type `InversableColorScheme`.

Reply:
We'll have to wait for @BPScott to come on line to be sure but I think this is ok. I think only items exported from `src/components/index.ts` are made public. So to make this public you'd need to export the type in that file as well

Reply:
@kyledurand you're right. This is fine as is. 

Feels like we're on the cusp of crosstalk where "public" means two different things: "Available to other components within this library" and "available as an export of `@shopify/polaris`".

`components/ThemeProvider/index.ts` reexports everything so it is available within the library internally by doing `import {InversableColorScheme} from 'BLAH/components/ThemeProvider';` as Marten has already done.

If you *really* need a type to be externally available, so consumers can do `import {InversableColorScheme} from '@shopify/polaris';` you'd need to export it from `src/components/index.ts` as Kyle says, but in this case (and most other cases) you don't shouldn't need to do that.

We don't need to expose most types publicly anyway, because the majority of the time consumers don't care about a specific type, but about how they're exposed from some props. They don't care about `InversableColorScheme`, they care about "Whatever type PopoverProps['colorScheme'] desires because I'm trying to render a Popover" (which just so happens to be `InversableColorScheme` at the moment).

Thus people can still get at this indirectly:

```
import {PopoverProps} from '@shopify/polaris'
const popoverColorScheme: PopoverProps['colorScheme'] = 'inverse'
```